### PR TITLE
chore(*): misc cleanup

### PR DESCRIPTION
**Pull request by SimonSchick**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [ ] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [ ] Have you added new tests to prevent regressions?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [ ] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/master/CONTRIBUTING.md)?

### Description of change

Many things:
In my last PR I missed updating the `test` folder so this PR also includes these.
Moved various lodash usage to the native counterpart (indexOf, includes, forEach, etc.).
Removed all usage of `self = this` (one exception) related `promise.bind()` use and `.then(function` use.
Update docs to suggest modern es6 usage.

I currently have some tests failing, trying to see if the occur on CI as well!

### CODE REVIEW

File: docs/models-definition.md [hunk 1]
```diff
@@ -724,37 +724,47 @@ sequelize.sync({ force: true, match: /_test$/ });
 Sequelize Models are ES6 classes. You can very easily add custom instance or class level methods.
```
Comment (sushantdhiman) [inline docs/models-definition.md]: For now lets remove all changes from this file (or other documentation) that introduce new class syntax, getters/setters support is still pending

**Comment by SimonSchick** [comment]
Still working on this PR, local testing keeps crashing due to deadlocks, not sure why :)

File: docs/models-definition.md (hunk 1 above)
Comment (SimonSchick) [reply docs/models-definition.md]: What do you mean by `getters/setters` support?

Comment (sushantdhiman) [reply docs/models-definition.md]: https://github.com/sequelize/sequelize/issues/9300

Comment (SimonSchick) [reply docs/models-definition.md]: That's sad, I've always used the `getter/setter` in the sequelize options, hence my confusion, maybe we should instead adjust the docs? I will revert it for now regardless.

**Comment by SimonSchick** [comment]
To clarify, none of my changes should have *any* breaking changes.
Sometimes CI seems to fail due to what I assume is a race condition.

File: package.json [hunk 2]
```diff
@@ -17,7 +17,7 @@
   "bugs": {
     "url": "https://github.com/sequelize/sequelize/issues"
   },
-  "main": "index.js",
+  "main": "./lib/sequelize.js",
```
[the remark this answers is not in the document]
Comment (SimonSchick) [reply package.json]: I tried that but I'd have to change _all_ test import paths.
This still avoids loading 1 proxy file when loading sequelize as a consumer.

[the remark this answers is not in the document]
Comment (sushantdhiman) [reply package.json]: Then revert this change please, there is negligible performance impact by loading one more file

[the remark this answers is not in the document]
Comment (SimonSchick) [reply package.json]: mhkay

**Comment by sushantdhiman** [comment]
Thanks @SimonSchick :+1:

**Pull request merged**